### PR TITLE
Support uppercase repository names with cosign.

### PR DESCRIPTION
**Pull request by mattmoor**
My previous PR didn't properly handle uppercase usernames (or repository names) when signing container images with `cosign`.

It seems that the `docker buildx --push` doesn't like this either, but it's passed the output of the `docker/metadata-action` which seems to lowercase things.

Fixes: https://github.com/actions/starter-workflows/issues/1293

Signed-off-by: Matt Moore <mattmoor@chainguard.dev>

thanks to @imjasonh for the suggestion!

## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _CI_ workflows, the workflow:**

- [x] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [x] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [x] Packaging workflows should run on `release` with `types: [ created ]`.
- [x] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `organization`: Name of the organization producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @bishal-pbMSFT @jhutchings1

**Review by jhutchings1 — APPROVED** [review]
Looks good to me

File: ci/docker-publish.yml [hunk 1]
```diff
@@ -90,4 +90,4 @@ jobs:
         # This step uses the identity token to provision an ephemeral certificate
         # against the sigstore community Fulcio instance, and records it to the
         # sigstore community Rekor transparency log.
-        run: cosign sign ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}@${{ steps.build-and-push.outputs.digest }}
+        run: cosign sign ${{ steps.meta.outputs.tags }}@${{ steps.build-and-push.outputs.digest }}
```
Comment (bishal-pdMSFT) [inline ci/docker-publish.yml:93]: Could it be a problem if `tags` points to a list of tags?

Comment (mattmoor) [reply ci/docker-publish.yml:93]: Given line 66, I don't think that's possible today, but it could be a problem as folks alter the workflow.

**Review by bishal-pdMSFT — APPROVED** [review]

**Comment by joshmgross** [comment]
👋  Looks like there are some merge conflicts, possibly from the comment change that went with #1295

**Comment by mattmoor** [comment]
I'll update this shortly

**Comment by mattmoor** [comment]
Ok, this is rebased!

**Comment by jhutchings1** [comment]
@mattmoor Do we still need this change to get merged in? I was surprised to see we never merged it back in December, but I guess holidays got in the way.

**Comment by mattmoor** [comment]
Yeah, I can rebase it, but it had fallen off my radar.  thanks for the ping.

**Comment by mattmoor** [comment]
Just noticed this still on my dashboard.  I'm happy to rebase, but it'd be great to coordinate with someone that can merge this to avoid spinning my wheels rebasing 🙏

**Comment by Phantsure** [comment]
@mattmoor should we go ahead with merge?

**Comment by mattmoor** [comment]
It's been a while, but I'm pretty sure this is good to go.  thanks!

**Pull request merged**